File: api/config/storage.js
/**
 * MinIO/S3ストレージの設定
 */

import { Client as Minio } from 'minio';
import { S3_CONFIG } from './secrets.js';

export function createMinioClient() {
  if (!S3_CONFIG.endpoint) {
    throw new Error('S3_ENDPOINT not configured');
  }

  const endpointUrl = S3_CONFIG.endpoint;
  const [host, port] = endpointUrl
    .replace('http://', '')
    .replace('https://', '')
    .split(':');

  return new Minio({
    endPoint: host,
    port: Number(port || 9000),
    useSSL: endpointUrl.startsWith('https://'),
    accessKey: S3_CONFIG.accessKey,
    secretKey: S3_CONFIG.secretKey,
  });
}

export function getS3Bucket() {
  return S3_CONFIG.bucket;
}

File: worker/app/storage.py
from minio import Minio

def make_minio(endpoint: str, access_key: str, secret_key: str) -> Minio:
    endpoint2 = endpoint.replace("http://", "").replace("https://", "")
    host, port = (endpoint2.split(":") + ["9000"])[:2]
    return Minio(
        f"{host}:{port}",
        access_key=access_key,
        secret_key=secret_key,
        secure=endpoint.startswith("https://"),
    )

def get_object_bytes(m: Minio, bucket: str, key: str) -> bytes:
    obj = m.get_object(bucket, key)
    return obj.read()
